Paws4Me/Paws4Me/Display All Adoptable Pets/View Models/PetViewModel.swift

```diff
@@ -0,0 +1,206 @@
+//
+//  PetViewModel.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/03/24.
+//
+
+import Foundation
+import UIKit
+
+// MARK: - ViewModel Delegate
+protocol PetViewModelDelegate: AnyObject {
+    func reloadView()
+    func show(error: String)
+}
+
+class GetAllPetDataViewModel {
+
+    // MARK: - Vars/Lets
+    private var petRepository: Repositable?
+    private weak var delegate: PetViewModelDelegate?
+    private var filteredPetObject: AdoptPet?
+    private var adoptPetObject: AdoptPet?
+    private var petCount: Int?
+    private var isSingleSearch = false
+    private var isFilterSearch = true
+    private var indexSinglePet: Int?
+    private var selectedGender: String = ""
+    private var animalType = ""
+
+    // MARK: - Constructor
+    init(repository: Repositable,
+         delegate: PetViewModelDelegate) {
+        self.petRepository = repository
+        self.delegate = delegate
+    }
+
+    // MARK: - Functions
+    // MARK: - Functions Fetch Json Object
+    func fetchPetDataResults() {
+        petRepository?.fetchPetDataResults(method: .GET, endpoint: Constants.adoptURL) { [weak self] result in
+            DispatchQueue.main.async {
+                switch result {
+                case .success(let petData):
+                    self?.adoptPetObject = petData
+                    self?.filteredPetObject = petData
+                    self?.delegate?.reloadView()
+                case .failure(let error):
+                    self?.delegate?.show(error: error.rawValue)
+                }
+            }
+        }
+    }
+
+    // MARK: - Tableview Functions
+    func getPetCount() -> Int {
+        if let pageCount = filteredPetObject?.page?.count {
```

Comment:
This can be simplified to: 

`return filteredPetObject?.page?.count ?? 0`

Reply:
This is much simpler. Thank you 👍🏻